lib/spack/spack/test/env.py

```diff
@@ -385,3 +378,40 @@ def test_can_add_specs_to_environment_without_specs_attribute(tmp_path, mock_pac
 
     assert len(env.user_specs) == 1
     assert env.manifest.pristine_yaml_content["spack"]["specs"] == ["a"]
+
+
+@pytest.mark.parametrize("filename", [ev.lockfile_name, "as9582g54.lock", "m3ia54s.json"])
+def test_initialize_from_lockfile(tmp_path, filename):
+    """Test that we can initialize an environment from a valid lockfile"""
```

Comment:
I didn't check style (edited from UI), but I would like to capture the use case in the docstring:
```suggestion
    """Some users have workflows where they store multiple lockfiles in the same directory, and pick one 
    of them to create an environment depending on external parameters e.g. while running CI jobs. This
    test ensures that Spack can create environments from lockfiles that are not necessarily named 'spack.lock'
    and can thus coexist in the same directory.
    """
```